packages/client/hmi-client/src/components/workflow/ops/model-config/tera-model-config.vue

```diff
@@ -630,6 +654,35 @@ const isRegNet = computed(() => modelType.value === AMRSchemaNames.REGNET);
 const isPetriNet = computed(() => modelType.value === AMRSchemaNames.PETRINET);
 const isStockFlow = computed(() => modelType.value === AMRSchemaNames.STOCKFLOW);
 
+const addIntervention = () => {
+	if (knobs.value.transientModelConfig.interventions) {
+		knobs.value.transientModelConfig.interventions.push({
+			id: uuidv4(),
+			name: '',
+			timestep: 1,
+			value: 1
+		});
+	} else {
+		knobs.value.transientModelConfig.interventions = [
+			{ id: uuidv4(), name: '', timestep: 1, value: 1 }
+		];
+	}
+};
+
+const deleteIntervention = (index: number) => {
+	if (knobs.value.transientModelConfig.interventions?.[index]) {
+		knobs.value.transientModelConfig.interventions.splice(index, 1);
+	}
+};
```

Comment:
This is a very strange setup, if you have ID then don't rely on index, if yo have index then don't rely on ID. Having both makes understanding the data structure more complicated as the rules for creation and deletion are mismatched.

Reply:
Thanks for the tip on updating too much and rerendering